index.bs

```diff
@@ -1657,52 +1724,51 @@ interface MLGraph {
 
     : <dfn>\[[implementation]]</dfn>
     ::
-        The underlying implemenation provided by the User Agent.
+        The underlying implemenation provided by the user agent.
 </dl>
 
 <dl dfn-type=method dfn-for=MLGraph>
     : <dfn>compute(inputs, outputs)</dfn>
     ::
-        Issue a compute request of the {{MLGraph}} given {{MLNamedInputs}} and optional {{MLNamedOutputs}}. The returned {{Promise}} resolves when the results in {{MLNamedOutputs}} are ready to be consumed.
+        Issue a compute request of the {{MLGraph}} given an input {{MLNamedTensors}} and optional output {{MLNamedTensors}}. The returned {{MLNamedTensors}} are the results ready to be consumed.
 
         <div algorithm=MLGraph.compute>
             **Called on:** {{MLGraph}} |this|.
 
             **Arguments:**
             <pre class=argumentdef for="MLGraph/compute(inputs, outputs)">
-                |inputs|: a {{MLNamedInputs}}. The data and optional dimensions of inputs for the compute request.
-                |outputs|: an optional {{MLNamedOutputs}}. The names and pre-allocated resources of required outputs for the compute request. Default to be an empty [=record=] which means that the compute request is for all outputs.
+                |inputs|: a {{MLNamedTensors}}. The data and optional dimensions of inputs for the compute request.
+                |outputs|: an optional {{MLNamedTensors}}. The names and pre-allocated resources of required outputs for the compute request. Default to be an empty [=record=] which means that the compute request is for all outputs.
             </pre>
 
-            **Returns:** {{Promise}}&lt;{{MLNamedOutputs}}&gt;. The dimensions and data of outputs returned by the compute request.
+            **Returns:** {{MLNamedTensors}}. The dimensions and data of outputs returned by the compute request.
 
-            1. Let |promise| be [=a new promise=].
             <!-- Validate inputs and outputs -->
-            1. If any of the following requirements are unmet, then [=reject=] |promise| with a {{TypeError}} and stop.
+            1. If any of the following requirements are unmet, then stop.
```

Comment:
Do we want to signal to the web developer the executing has stopped?

There's a leftover promise reference on L1801 that should be either removed or repurposed for signaling this, this one:

reject promise with an OperationError and stop.